chat-server/src/routes/conversations/addMessageToConversation.ts

```diff
@@ -203,24 +237,56 @@ export async function getContentForText({
     text,
     userIp: ipAddress,
   });
-  return store.findNearestNeighbors(embedding, findNearestNeighborsOptions);
+  return await store.findNearestNeighbors(
+    embedding,
+    findNearestNeighborsOptions
+  );
+}
+
+interface AddMessagesToDatabaseParams {
+  conversationId: ObjectId;
+  userMessageContent: string;
+  assistantMessageContent: string;
+  conversations: ConversationsServiceInterface;
+}
+export async function addMessagesToDatabase({
+  conversationId,
+  userMessageContent,
+  assistantMessageContent,
+  conversations,
+}: AddMessagesToDatabaseParams) {
+  // TODO: consider refactoring addConversationMessage to take in an array of messages.
+  // Would limit database calls.
+  const userMessage = await conversations.addConversationMessage({
+    conversationId,
+    content: userMessageContent,
+    role: "user",
+  });
+  const assistantMessage = await conversations.addConversationMessage({
+    conversationId,
+    content: assistantMessageContent,
+    role: "assistant",
+  });
+  return { userMessage, assistantMessage };
 }
 
 export interface GenerateFurtherReadingParams {
   chunks: EmbeddedContent[];
+  hasHeading?: boolean;
 }
 export function generateFurtherReading({
   chunks,
+  hasHeading = true,
 }: GenerateFurtherReadingParams) {
   if (chunks.length === 0) {
     return "";
   }
-  const heading = "\n\nArticles Referenced:\n";
+  const heading = hasHeading ? "\n\nFurther Reading:\n" : "";
   const uniqueLinks = Array.from(new Set(chunks.map((chunk) => chunk.url)));
 
   const linksText =
     uniqueLinks.reduce((acc, link) => {
-      const linkListItem = `${link}\n`;
+      const linkListItem = `[${link}](${link}?tck=docs_chatbot)\n`;
```

Comment:
V. nice.

Btw... we might want to start using [URL](https://nodejs.org/api/url.html) more around the code to ensure correct URL building.

Reply:
good point. will look at refactoring here if it's not too hard. though this logic is going to move to the FE soon, so don't want to spend too much time on it.

Reply:
refactored to use this.